staging/src/k8s.io/legacy-cloud-providers/aws/aws.go

```diff
@@ -5010,6 +5013,19 @@ func (c *Cloud) nodeNameToProviderID(nodeName types.NodeName) (InstanceID, error
 	return KubernetesInstanceID(node.Spec.ProviderID).MapToAWSInstanceID()
 }
 
+func checkMixedProtocol(ports []v1.ServicePort) error {
+	if len(ports) == 0 {
+		return nil
+	}
+	firstProtocol := ports[0].Protocol
+	for _, port := range ports {
```

Comment:
1. how about ```for _, port := range ports[1:]``` since ports is at least one element from `if len(ports) == 0`.
2. will APIServer automatically fill the protocol to "TCP" if not specified? from the spec seems it's default to tcp, but want to verify that. e.g. 
````
ports:
   - port: 80
      protocol: TCP
  -  port: 8080
      protocol:
````

Reply:
1. Updated
2. Verified if protocol is blank, it defaults to TCP